Issue #458: Add error handling when deleting firmware with associated deployments

chazwatkins opened the pull request:
Closes #458 

Adds an error flash message when trying to delete a firmware that has associated deployments.

jjcarstens commented on apps/nerves_hub_www/lib/nerves_hub_www_web/controllers/firmware_controller.ex:
```diff
@@ -93,11 +93,18 @@ defmodule NervesHubWWWWeb.FirmwareController do
+          |> put_flash(:info, "Firmware successfully deleted")
+          |> redirect(to: Routes.firmware_path(conn, :index, org.name, product.name))
+
+        {:error, %Ecto.Changeset{errors: [deployments: _]}} ->
```
Since this is a pretty specific match, we probably want one more to catch _all_ potential errors.

chazwatkins replied:
I've added the `catch-all` condition

jjcarstens commented on apps/nerves_hub_www/lib/nerves_hub_www_web/controllers/firmware_controller.ex:
```diff
@@ -93,11 +93,18 @@ defmodule NervesHubWWWWeb.FirmwareController do
-         :ok <- Firmwares.delete_firmware(firmware) do
-      conn
-      |> put_flash(:info, "Firmware successfully deleted")
-      |> redirect(to: Routes.firmware_path(conn, :index, org.name, product.name))
```
This is a super nit-pick, but we can probably just keep the `with` statement as is and add associated matching `else` clauses for the errors:

```elixir
with {:ok, firmware} <- Firmwares.get_firmware_by_product_and_uuid(product, uuid),
          :ok <- Firmwares.delete_firmware(firmware) do
  ...
else
  {:error, %Ecto.Changeset{errors: [deployments: _]}} ->
    ..
  {:error, e} ->
    ..
end
```

chazwatkins replied:
Reverted to the `with` as requested and added the `else` pattern matching for both deployments and catch-all.

chazwatkins commented:
@jjcarstens Thanks for taking the time to review the contribution.

The requested changes have been implemented

1. Revert from `case` statement within the `with` to `with-else` in `nerves_hub_www_web/controllers/firmware_controller.ex`
2. Added catch-all error match that returns a generic error.
3. Added same error handling in `nerves_hub_api_web/controllers/firmware_controller.ex`
